### PR TITLE
[GFC-858][AK]new endPoint => for users for groupId where list[UserDet…

### PR DESCRIPTION
**Pull request by arkowalski**
…ailsData].head != (_.credentialRole == User) assign Enrolment using ES11

### CODE REVIEW

File: app/uk/gov/hmrc/eeittadminfrontend/connectors/UserDetailsConnector.scala [hunk 1]
```diff
@@ -37,4 +37,16 @@ object UserDetailsConnector {
...
       }
+
+  def userIdsByGroupIdNotHeadAdmin(
+    groupId: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[List[String]] =
+    WSHttp
+      .GET[HttpResponse](s"${sc.baseUrl("user-details")}/user-details/group-identifier/$groupId")
+      .map { response =>
+        val a = response.json.as[List[UserDetailsData]]
+
+        val alreadyDoneAdmin = a.filterNot(_.credentialRole != "User")
```
Comment (lancewalton) [inline app/uk/gov/hmrc/eeittadminfrontend/connectors/UserDetailsConnector.scala]: Use type safe equality?

File: test/uk/gov/hmrc/eeittadminfrontend/controllers/DeltaControllerSpec.scala [hunk 2]
```diff
@@ -198,7 +203,7 @@ class DeltaControllerSpec
         .withSession("token" -> "someGoogleAuthenticationToken")
```
Comment (lancewalton) [inline test/uk/gov/hmrc/eeittadminfrontend/controllers/DeltaControllerSpec.scala]: There is a lot of duplication in this file

File: app/uk/gov/hmrc/eeittadminfrontend/connectors/EnrolmentStoreProxyConnector.scala [hunk 3]
```diff
@@ -77,6 +77,16 @@ object EnrolmentStoreProxyConnector {
       s"$enrolmentStoreProxyBaseUrl/groups/$groupId/enrolments/${TaxEnrolment.enrolmentKey(identifiers)}"
     )
 
+  // ES11
+  def allocateEnrolmentForUser(userId: String, identifiers: List[Identifier], verifiers: List[Verifier])(
+    implicit hc: HeaderCarrier,
+    ec: ExecutionContext): Future[HttpResponse] =
+    WSHttp
+      .POST(
+        s"$enrolmentStoreProxyBaseUrl/users/$userId/enrolments/${TaxEnrolment.enrolmentKey(identifiers)}",
```
[the remark this answers is not in the document]
Comment (arkowalski) [reply app/uk/gov/hmrc/eeittadminfrontend/connectors/EnrolmentStoreProxyConnector.scala]: has been successfully used before

File: test/uk/gov/hmrc/eeittadminfrontend/controllers/DeltaControllerSpec.scala (hunk 2 above)
Comment (arkowalski) [reply test/uk/gov/hmrc/eeittadminfrontend/controllers/DeltaControllerSpec.scala]: purely for time efficiency as the repository is to be decommissioned soon and the urgency for the particular task

**Pull request merged**